app/controllers/items_controller.rb

```diff
@@ -0,0 +1,76 @@
+require 'net/http'
+
+class ItemsController < ApplicationController
+
+	def new
+		@item = Item.new
+	end
+
+	def create
+		@item = Item.new
+
+		url = item_params[:item_url]
+
+		if !validate_url(url)
+			flash[:error] = "Please enter a valid Ikea product URL or Article Number!"
+			render :new and return
+		end
+
+		create_item(@item, url, item_params)
```

Comment:
Going to leave this for a later refactoring; this and a couple other refactors to the controller ended up causing all sorts of weird issues.